lib/server/routes/buckets.js

```diff
@@ -1413,6 +1416,361 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
+      return next(new errors.InternalError(err.message));
+    }
+
+    if (!bucket) {
+      return next(new errors.NotFoundError('Bucket not found'));
+    }
+
+    async.mapLimit(
+      shards,
+      1,
+      (shard, nextShard) => {
+        const { uuid, hash } = shard;
+
+        Upload.findOne({
+          uuid
+        }, (err, upload) => {
+          if (err) {
+            return nextShard(err);
+          }
+
+          if (!upload) {
+            return nextShard(new errors.NotFoundError('Upload not found'));
+          }
+
+          const uploadObject = upload.toObject();
+          const { _id, uuid, index, data_size, contracts } = uploadObject;
+
+          if (!uuid) {
+            return nextShard(new errors.BadRequestError('Shard without UUID'));
+          }
+
+          const shardToCreate = new Shard({
+            index,
+            data_size,
+            uuid,
+            hash,
+            contracts: contracts.map(({ nodeID, contract }) => ({
+              nodeID,
+              contract: {
+                ...contract,
+                data_hash: hash
+              }
+            })),
+          });
+
+          shardToCreate.save((err, savedShard) => {
+            if (err) {
+              return nextShard(err);
+            }
+
+            createdShards.push({
+              index: uploadObject.index,
+              size: uploadObject.data_size,
+              _id: savedShard._id,
+            });
+
+            async.eachLimit(
+              contracts,
+              1,
+              (contract, cb) => {
+                const mirrorToCreate = new Mirror({
+                  isEstablished: true,
+                  shardHash: hash,
+                  contact: contract.nodeID,
+                  contract,
+                  token: '',
+                  created: new Date()
+                });
+
+                mirrorToCreate.save(cb);
+              },
+              (err) => {
+                if (err) {
+                  return nextShard(err);
+                }
+
+                nextShard(null, upload);
+              }
+            );
+          });
+        });
+      },
+      (err, uploads) => {
+        if (err) {
+          return next(err);
+        }
+
+        const bucketEntryToCreate = new BucketEntry({
+          bucket: idBucket,
+          index,
+          created: new Date(),
+          shards: createdShards,
+        });
+
+        bucketEntryToCreate.save((err, savedBucketEntry) => {
+          if (err) {
+            return next(err);
+          }
+
+          const savedBucketEntryObject = savedBucketEntry.toObject();
+          res.status(200).send(savedBucketEntryObject);
+
+          async.eachLimit(
+            uploads,
+            1,
+            (upload, cb) => {
+              upload.remove((err) => {
+                if (err) {
+                  log.error('Error deleting Upload model with id: %s, error: %s', upload._id, err.message);
+
+                  return;
```

Comment:
use `cb(err)` and print the log in the eachLimit callback. Thus, we can avoid letting `eachLimit` stalled

Reply:
Fixed